src/Core/Resolvers/BaseQueryStructure.cs

```diff
@@ -111,13 +120,27 @@ public class BaseQueryStructure
         public virtual string MakeDbConnectionParam(object? value, string? paramName = null)
         {
             string encodedParamName = GetEncodedParamName(Counter.Next());
-            if (!string.IsNullOrEmpty(paramName))
+            if (string.IsNullOrEmpty(paramName))
             {
-                Parameters.Add(encodedParamName, new(value, GetUnderlyingSourceDefinition().GetDbTypeForParam(paramName)));
+                Parameters.Add(encodedParamName, new(value));
             }
             else
             {
-                Parameters.Add(encodedParamName, new(value));
+                SourceDefinition sourceDefinition = GetUnderlyingSourceDefinition();
+                DbType? dbTypeForParam = sourceDefinition.GetDbTypeForParam(paramName);
+                Type systemTypeForParam = sourceDefinition.GetSystemTypeForParam(paramName);
+
+                if (dbTypeForParam is null || ApiType is ApiType.GraphQL && systemTypeForParam == typeof(DateTime))
```

Comment:
nit: add additional brackets around conditions (based on evaluation order). Though it is possible to understand based on the logic below, it would help with readability.